### PR TITLE
Convert to typescript

### PR DESCRIPTION
**Pull request by thgreasi**
That's on top of #20 and will merge them together as v3.
Marked as major since the typings change.

Connects-to: #19
Change-type: major
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * @example
+ * balena = require('balena-sdk')
+ * deviceStatus = require('balena-device-status')
+ *
+ * balena.models.device.get('9174944').then (device) ->
+ * 	deviceStatus.getStatus(device).then (status) ->
+ * 		console.log(status.key)
+ * 		console.log(status.name)
+ */
+export const getStatus = (device: Device): Status => {
```
Comment (sradevski) [inline lib/index.ts]: Do we need the explicit return type?

File: lib/index.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	{ key: status.CONFIGURING, name: 'Configuring' },
+	{ key: status.UPDATING, name: 'Updating' },
+	{ key: status.OFFLINE, name: 'Offline' },
+	{ key: status.POST_PROVISIONING, name: 'Post Provisioning' },
+	{ key: status.INACTIVE, name: 'Inactive' },
+];
+
+const statusesByKey = keyBy(statuses, 'key');
+
+export interface IHaveProgressAndStatus {
```
Comment (sradevski) [inline lib/index.ts]: NIT: Maybe we can group all the interface definitions together.

**Review by sradevski — APPROVED** [review]
LGTM, just some NIT comments.

File: lib/index.ts (hunk 1 above)
Comment (thgreasi) [reply lib/index.ts]: Actually we don't, but I would prefer to be sure that this has the exact intended type in the built typings.
Moreover, this way we will notice it early if by accident we add a return statement that alters the return type.

File: lib/index.ts (hunk 2 above)
Comment (thgreasi) [reply lib/index.ts]: Let me actually move all of them in a different file

**Pull request merged**